### PR TITLE
Disable button if tests can't be run

### PR DESCRIPTION
**Pull request by adriana-alter**
Resolves https://jira.nypl.org/browse/SIMPLY-2493

It might be a better idea to hide the button altogether; opinions?

Screen Shot 2020-02-14 at 1 54 23 PM

### CODE REVIEW

File: src/components/SelfTests.tsx [hunk 1]
```diff
@@ -70,6 +70,7 @@ export class SelfTests extends React.Component<SelfTestsProps, SelfTestsState> {
   render(): JSX.Element {
     const integration: ServiceData = this.state.mostRecent;
     const selfTestException = integration.self_test_results && integration.self_test_results.exception;
+    const cannotRunTests = selfTestException?.search(/[before you can run self tests]/) >= 0;
```
Comment (EdwinGuzman) [inline src/components/SelfTests.tsx]: Is there a better way to set this up so it's not doing a regex on a string? Perhaps make the server return a flag that says that service cannot run self-tests?

Comment (adriana-alter) [reply src/components/SelfTests.tsx]: I will investigate!  Just so I understand, though...what's the problem with doing it as a regex?

Comment (adriana-alter) [reply src/components/SelfTests.tsx]: Update: turns out the server IS already returning a flag (I just didn't see it before)!

Comment (EdwinGuzman) [reply src/components/SelfTests.tsx]: I don't like relying on strings to do checks... seems like sturdy.

**Pull request merged**